lib/config/fileconf.go

```diff
@@ -1695,7 +1695,7 @@ type Databases struct {
 	// Databases is a list of databases proxied by the service.
 	Databases []*Database `yaml:"databases"`
 	// ResourceMatchers match cluster database resources.
-	ResourceMatchers []ResourceMatcher `yaml:"resources,omitempty"`
+	ResourceMatchers []DatabaseResourceMatcher `yaml:"resources,omitempty"`
```

Comment:
nit: 
Do you see any cons of adding the `AWS ResourceMatcherAWS` member to `ResourceMatcher` without adding a `DatabaseResourceMatcher`  ?. 
The `AWS` member in ResourceMatcher can be marked as optional and reused between kube app access in the future.

Reply:
I originally added a new type to ensure that other resource matchers will throw error if `assume_role_arn` is specified. 

I removed the `DatabaseResourceMatcher` in https://github.com/gravitational/teleport/pull/28039/commits/4a9acaeb2bee4e5142d369943611afcca5491c77, and explicitly throwing errors for other two types.